src/resolvers/User.js

```diff
@@ -0,0 +1,279 @@
+import {
+  ApolloError,
+  AuthenticationError,
+  UserInputError,
+} from "apollo-server-express";
+import validator from "email-validator";
+import jwt from "jsonwebtoken";
+import mongoose from "mongoose";
+import bcrypt from "bcrypt";
+
+import { User } from "../models";
+import {
+  validateLogIn,
+  validateSignUp,
+  validateRole,
+} from "../utils/validator";
+import { passwordEncryptor } from "../utils/encryption";
+
+const generateToken = (user) => {
+  return jwt.sign(
+    {
+      id: user._id,
+      type: user.type,
+      employeeType: user.employeeType,
+    },
+    process.env.SECRET_KEY,
+    {
+      expiresIn: "31536000000", // 1 year in millis
+    }
+  );
+};
+
+const UserResolver = {
+  Query: {
+    users: async (root, { filter }, context, info) => {
+      /**
+       * - check null filter
+       * - parse filter if not null
+       * - find by filter
+       */
+
+      if (filter === null) {
+        throw new UserInputError("No filter provided, use {} to get all");
+      }
+
+      /**
+       * Generate filter fron string
+       */
+      const parsedFilter = JSON.parse(filter);
+
+      /**
+       * Search with filter
+       */
+      const filteredUsers = await User.find(parsedFilter);
+
+      return filteredUsers;
+    },
+    user: async (root, { id }, context, info) => {
+      /**
+       * - check null id
+       * - validate mongoose id
+       * - find by id
+       */
+      if (id) {
```

Comment:
else will throw an exception?
or just return 200 with empty response

Reply:
> @LUXIANZE
> Few items suggest here:
> 
> * [ ]  gitignore package-lock.json
> * [ ]  utils/index.js is empty should remove
> * [ ]  UserInputValidator (validate signUp) purpose when there is a typeDefs

- I think package-lock.json should remain as mentioned in [this post](https://stackoverflow.com/a/44210813)
- the other 2 I agree